hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHAInvocationHandler.java

```diff
@@ -50,16 +50,20 @@ public SCMHAInvocationHandler(final RequestType requestType,
     this.requestType = requestType;
     this.localHandler = localHandler;
     this.ratisHandler = ratisHandler;
-    ratisHandler.registerStateMachineHandler(requestType, localHandler);
+    if (ratisHandler != null) {
+      ratisHandler.registerStateMachineHandler(requestType, localHandler);
+    }
   }
 
   @Override
   public Object invoke(final Object proxy, final Method method,
                        final Object[] args) throws Throwable {
     try {
       long startTime = Time.monotonicNow();
-      final Object result = method.isAnnotationPresent(Replicate.class) ?
-          invokeRatis(method, args) : invokeLocal(method, args);
+      final Object result =
```

Comment:
Please add a Preconditions.checkNotNull for ratisHandler at invokeRatis too. Thanks!